### DISCUSSION TITLE
How to log and track user data access ?

### DISCUSSION THREAD

**Discussion by kleysonr** [codenotary/immudb#1928] [Q&A]
Hi.

Due to some regulations I need to log and track all the access done over our sensitive data by someone.

I would like to try Immudb but not sure yet how to model my database.

The data access will stored and indexed by the user ID that was accessed. For example

```
{
  uid: "123123",
  data: {
    "accessed_by": {
      uid: "555666",
      name: "Jane"
    },
    timestamp: 12345323343,
    data_accessed: ["name", "birth_date", "address", "mobile_number"]
  }
}
```

So the uid: "123123" might be accessed by different people and accessing different kind of data.

During some auditing process, I need to show who and what was accessed regarding the uid: "123123".

Should I create a database for each uid ?
Should I use a single database ?

What would be the best architectural design approach to log all those access, considering I huge amount of users and data access logged ?

Best Regards.
Kleyson Rios.

**Comment by Razikus** [comment]
Hello

In my opinion index on uid field is enough
You dont need to create database for each uid

Most probably the best to choose is document model
https://docs.immudb.io/master/develop/document/api.html with index on uid 

So the query to check what uid 123123 is accessing is relatively simple

```
curl -X 'POST' \
'http://localhost:8080/api/v2/collection/mycollection/documents/search' \
-H "sessionID: $sessionid" \
-H 'accept: application/json' \
-H 'Content-Type: application/json' \
-d '{
  "query": {
    "expressions": [
      {
        "fieldComparisons": [
          {
            "field": "uid",
            "operator": "EQ",
            "value": "123123"
          }
        ]
      }
    ]
  },
  "page": 1,
  "pageSize": 10
}'
```
